src/main/java/com/teragrep/blf_01/TokenScan.java

```diff
@@ -169,36 +169,34 @@ private ByteBuffer extendBuffer(ByteBuffer byteBuffer, int size) {
     // -----
 
     Delimiter match(Delimiters delimiters, ByteBuffer matchBuffer) {
-        Delimiter rv = new Delimiter();
-
-
-        for (Delimiter delimiter : delimiters.getDelimiters()) {
-            // debug
-
-            if (matchBuffer.equals(delimiter.delimiterBuffer)) {
-                // has match
-                rv = delimiter;
-            }
-        }
+        Delimiter rv = getOrCreateDelimiter(delimiters, matchBuffer);
 
         if (rv.isStub) {
             // saerch smaller delimiter
             // TODO get delimiter by size
-            Delimiter sub = new Delimiter();
             if (matchBuffer.limit() > 1) {
                 ByteBuffer sliceBuffer = matchBuffer.slice();
                 ByteBuffer subMatchBuffer = (ByteBuffer) sliceBuffer.limit(matchBuffer.limit() - 1);
-                sub = match(delimiters, subMatchBuffer);
-            }
-            if (!sub.isStub) {
-                if (sub.delimiterBuffer.capacity() > rv.delimiterBuffer.capacity()) {
-                    rv = sub;
+                Delimiter sub = match(delimiters, subMatchBuffer);
+                if (!sub.isStub) {
+                    if (sub.delimiterBuffer.capacity() > rv.delimiterBuffer.capacity()) {
```

Comment:
this is kind of extra, because if rv.isStub == true -> rv.delimiterBuffer.capacity is always 0